app/models/submission.rb

```diff
@@ -7,6 +7,7 @@ class Submission < ApplicationRecord
 
   has_many :files, dependent: :nullify, class_name: 'SubmissionFile'
   has_many :entries, dependent: :nullify, class_name: 'SubmissionEntry'
+  has_many :invoices, dependent: :nullify, class_name: 'SubmissionInvoice'
```

Comment:
This is a good point, I went with has_many, because as it stood, there is indeed nothing stopping multiple being created, similar to files. I guess that would be bad thinking about it.

Although looking through the code, I think AASM would stop more than one being generated, as we can't transition a submission to completed twice?

Reply:
True, although the state machine wouldn't prevent the job re-running for whatever reason. Perhaps a guard clause that raises an exception if the `submission.invoice.exists?` at the top of the job would suffice.

Reply:
Also files is that way because we do expect and plan to add support for submissions with multiple files. this is a requirement because some suppliers have submissions that have more rows than Excel can actually support (e.g. the energy framework), so they will need to make a submission with multiple files.